### PR TITLE
Accommodate timezone based on zip code in consult flow

### PR DESCRIPTION
**Pull request by craigspaeth**
This picks different content modules for the billing screen based on if the zip-code is in California. That way we can communicate to CA-based clients that they can use our insurance partners.

image

It also renders timezone in the scheduler and determines based on zipcode or guessed from IP address (using `moment().tz().guess()`).

Screen Shot 2019-07-09 at 6 37 38 PM

### CODE REVIEW

File: apps/onboarding/lib/view-model.js [hunk 1]
```diff
@@ -69,6 +70,94 @@ export const initialState = {
   errorMsg: null
 }
 
+export const consultContentQuery = ({
```
Comment (craigspaeth) [inline apps/onboarding/lib/view-model.js]: This is a _Scouts Rule_ thing where I moved the duped consult queries b/t class and regular consults into the view model.

File: apps/onboarding/lib/view-model.js [hunk 2]
```diff
@@ -433,6 +522,11 @@ export const nextStep = state => ({
   errorMsg: null
 })
 
+export const billingContent = (state, content) =>
+  zipcodes.lookup(state.person.zipCode).state === 'CA'
```
Comment (craigspaeth) [inline apps/onboarding/lib/view-model.js]: This is the bit that uses [zipcodes](https://www.npmjs.com/package/zipcodes) to determine if they're based in CA and pick a different content module if so.

File: lib/availability-picker/index.js [hunk 3]
```diff
@@ -114,6 +123,7 @@ const Availabilities = ({
 )
 
 export default ({
+  timezone = moment.tz.guess(),
```
Comment (craigspaeth) [inline lib/availability-picker/index.js]: Here we accept passing in the timezone and default to guessing based on the computer's IP address.

File: lib/availability-picker/index.js [hunk 4]
```diff
@@ -56,30 +58,35 @@ const AvailabilitiesList = ({
...
-                formatDatetime(selectedAvailability.startTime)
-              ? 'is-selected'
-              : ''
-          )}
-        >
-          {moment(availability.startTime).format('h:mm a')}
-        </div>
-      ))}
+      {availabilities.map(availability => {
+        const tzStartTime = momentTz(availability.startTime).tz(timezone)
```
Comment (craigspaeth) [inline lib/availability-picker/index.js]: Here we adjust the availabilities to the timezone.

File: apps/onboarding/components/schedule-consult.js [hunk 5]
```diff
@@ -68,6 +69,7 @@ export default props => {
       ),
       body: (
         <AppointmentForm
+          timezone={zipToTz.lookup(state.person.zipCode)}
```
Comment (craigspaeth) [inline apps/onboarding/components/schedule-consult.js]: Here we pass in the timezone based on the entered zip code. We're also using this library in appointment reminder/confirmations—isomorphic JS FTW!

File: apps/onboarding/lib/view-model.js [hunk 6]
```diff
@@ -517,6 +613,16 @@ const trackStepAndScrollUp = state => {
   window.scrollTo({ top: 0 })
 }
 
+export const confirmParagraph = (state, content) => {
```
Comment (craigspaeth) [inline apps/onboarding/lib/view-model.js]: This injects the "See you {{onDate}}" variable but now corrected to the client's timezone and including the TZ code.

File: apps/onboarding/components/schedule-consult.js (hunk 5 above)
Comment (lionsdwarf) [reply apps/onboarding/components/schedule-consult.js]: Rock on!! 🎸

File: lib/availability-picker/index.js (hunk 4 above)
Comment (lionsdwarf) [reply lib/availability-picker/index.js]: 👍

File: lib/availability-picker/index.js (hunk 3 above)
Comment (lionsdwarf) [reply lib/availability-picker/index.js]: Nice method

**Pull request merged**